dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -165,21 +173,44 @@ public Object visitThread(LitmusCParser.ThreadContext ctx) {
         return null;
     }
 
+    @Override
+    public Object visitOpenCLThreadScope(LitmusCParser.OpenCLThreadScopeContext ctx) {
+        int wgID = ctx.scopeID(0).id;
+        int devID = ctx.scopeID(1).id;
+        programBuilder.setOrCreateScopedThread(Arch.OPENCL, currentThread, devID, wgID);
+        return null;
+    }
+
     @Override
     public Object visitThreadArguments(LitmusCParser.ThreadArgumentsContext ctx){
         if(ctx != null){
-            for(LitmusCParser.VarNameContext varName : ctx.varName()){
-                // TODO: Possibly parse attributes/type modifiers (const, atomic, ...)
-                //  For now, herd7 also seems to ignore most modifiers, in particular the atomic one.
-                String name = varName.getText();
-                MemoryObject object = programBuilder.getOrNewMemoryObject(name);
-                Register register = programBuilder.getOrNewRegister(scope, name, archType);
-                programBuilder.addChild(currentThread, EventFactory.newLocal(register, object));
+            for(LitmusCParser.ThreadArgumentContext threadArgumentContext : ctx.threadArgument()){
+                threadArgumentContext.accept(this);
             }
         }
         return null;
     }
 
+    @Override
+    public Object visitThreadArgument(LitmusCParser.ThreadArgumentContext ctx) {
+        // TODO: Possibly parse attributes/type modifiers (const, ...)
+        //  For now, herd7 also seems to ignore most modifiers, in particular the atomic one.
+        String name = ctx.varName().getText();
+        MemoryObject object = programBuilder.getOrNewMemoryObject(name);
+        Register register = programBuilder.getOrNewRegister(scope, name, archType);
+        boolean atomicity = ctx.pointerTypeSpecifier().atomicTypeSpecifier() != null
+                || ctx.pointerTypeSpecifier().basicTypeSpecifier().AtomicInt() != null;
+        if (!atomicity) {
+            object.addFeatureTag(C11.NON_ATOMIC_LOCATION);
+        }
+        if (isOpenCL && ctx.pointerTypeSpecifier().openCLSpace() != null) {
+            object.addFeatureTag(ctx.pointerTypeSpecifier().openCLSpace().space);
+        }
+        programBuilder.setReg2LocMap(register, object);
+        programBuilder.addChild(currentThread, EventFactory.newLocal(register, object));
```

Comment:
This is what I was suggesting above. Why do we even need the reg2Loc map if we are already doing this?